changelog/27881.txt

```diff
@@ -0,0 +1,4 @@
+```release-note:bug
+sys: On mount tune with `plugin_version`, set MountEntry.Type to the plugin name, Vault post-v1.0.0's scheme,
+if the mount was created in Vault pre-v1.0.0, which sets "plugin" to MountEntry.Type.
```

Comment:
nit: this is a bit technical, maybe we could make it a little higher level? Something like...

```suggestion
sys/plugin: Fix a bug where external plugins that were registered before Vault v1.0.0 could not be upgraded to be versioned plugins.
```

Reply:
fixed in [0f79f33](https://github.com/hashicorp/vault/pull/27881/commits/0f79f33b8e39e2f759fdb4441a9f3e7609361b99)